File: 645 - Set Mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    let res = [-1, -1];
    let counts = new Array(nums.length);
    counts = counts.fill(0);
    for (let index = 0; index < nums.length; index++) {
        counts[nums[index] - 1]++;
    }
    for (let index = 0; index < counts.length; index++) {
        if (counts[index] == 2) res[0] = index + 1;
        if (counts[index] == 0) res[1] = index + 1;
    }
    return res;
};

File: 645 - Set Mismatch.java
import java.util.Arrays;

class Solution {
    public int[] findErrorNums(int[] nums) {
        int[] res = {-1, -1};
        int[] counts = new int[nums.length];
        Arrays.fill(counts, 0);
        for (int i = 0; i < nums.length; i++) {
            counts[nums[i] - 1]++;
        }
        for (int i = 0; i < counts.length; i++) {
            if (counts[i] == 2) res[0] = i + 1;
            if (counts[i] == 0) res[1] = i + 1;
        }
        return res;
    }
}